src/hotspot/share/code/nmethod.cpp

```diff
@@ -3488,6 +3488,9 @@ void nmethod::decode2(outputStream* ost) const {
   if (use_compressed_format && ! compressed_with_comments) {
     const_cast<nmethod*>(this)->print_constant_pool(st);
 
+    st->bol();
+    st->cr();
+    st->print_cr("Loading hsdis library failed, so undisassembled code is printed in the below [MachCode] section");
```

Comment:
Some comments on the message:
 - I don't think we should use `[MachCode]` inside the square brackets apart from marking the start of a section. Otherwise, tools parsing a hs_err file incorrectly identify the start of the MachCode section.
 - Personally, I find the language of the message a bit clunky. Here are a few suggestions I would personally prefer:
   `Loading hsdis library failed, unable to show disassembled code in MachCode section` or `Note: Unable to display disassembled code because loading of hsdis library failed.`.

Reply:
I do happen to dislike "undisassembled", but I was not aware of its usage in other places in the codebase. I'm fine with the message now.